test/test_nn.py

```diff
@@ -5609,6 +5609,19 @@ def test_cosine_similarity(self):
         out = F.cosine_similarity(input.to(torch.int8), input, dim=-1)
         self.assertEqual(out, 1.)
 
+        # Check broadcasting #109333
+        a = torch.ones(2, 3, dtype=torch.float)
+        b = torch.ones(1, 1, dtype=torch.float)
+        out = F.cosine_similarity(a, b)
+        self.assertEqual(out, torch.ones(2, dtype=torch.float))
+
+        a = torch.ones(2, 3, dtype=torch.float)
+        b = torch.ones(1, dtype=torch.float)
+        out = F.cosine_similarity(a, b)
+        self.assertEqual(out, torch.ones(2, dtype=torch.float))
```

Comment:
@kurtamohler you were looking at removing default dtypes from the tests. 
FYI, here, I created a `torch.ones(2)` and it defaulted to `float64`. I didin't look any further tho.

Reply:
I removed `set_default_dtype` calls that have a lasting effect on the state before and after each test is run. But a lot of tests, including `test_cosine_similarity`, failed after I changed the global default dtype from `torch.double` to `torch.float`. Rather than attempt to make all of those tests pass with `torch.float`, I opted to temporarily change the default dtype back to `torch.double`--in this case, by wrapping the test function with the decorator `@set_default_dtype(torch.double)`. The wrapper is guaranteed to restore the original default dtype after the test runs

Reply:
Ah, I had missed that decorator. Thank you for the explanation!